api_app/analyzers_manager/observable_analyzers/dns/dns_malicious_detectors/spamhaus_wqs.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+import requests
+
+from api_app.analyzers_manager import classes
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+from ..dns_responses import malicious_detector_response
+
+logger = logging.getLogger(__name__)
+
+
+class SpamhausWQS(classes.ObservableAnalyzer):
+    url: str = "https://apibl.spamhaus.net/lookup/v1"
+    _api_key: str = None
+
+    def update(self):
+        pass
+
+    def run(self):
+        headers = {"Authorization": f"Bearer {self._api_key}"}
+
+        if self.observable_classification == self.ObservableTypes.DOMAIN.value:
+            if len(self.observable_name.split(".")) > 2:
```

Comment:
may I ask you why this choice?

Reply:
Mb, I was thinking something stupid. Corrected.